dashboard/src/components/Catalog/Catalog.tsx

```diff
@@ -141,8 +168,8 @@ function Catalog(props: ICatalogProps) {
   // Only one search filter can be set
   const searchFilter = propsFilter[filterNames.SEARCH]?.toString() || "";
   useEffect(() => {
-    fetchCharts(cluster, namespace, repo, searchFilter);
-  }, [fetchCharts, cluster, namespace, repo, searchFilter]);
+    fetchCharts(cluster, namespace, currentRepo, searchFilter);
```

Comment:
why do you need to store a separate state variable? won't it work simply using the filters already in the state? Something like:

```js
const reposFilter = filters[filterNames.REPOS]?.join(",") || "";
 useEffect(() => {
    fetchCharts(cluster, namespace, reposFilter, searchFilter);
  }, [fetchCharts, cluster, namespace, reposFilter, searchFilter]);
```

Reply:
You're right, if we already have the `filters` in the state it doesn't really make sense. I wrongly thought that, perhaps, `reposFilter`  was not always referring to the current value,